src/FSM/States/Downloading.cpp

```diff
@@ -49,8 +51,15 @@ namespace sua {
             ctx.desiredState.downloadBytesDownloaded    = std::stoll(payload.at("downloaded"));
             ctx.desiredState.downloadProgressPercentage = std::stoi(payload.at("percentage"));
 
+            const auto now            = std::chrono::system_clock::now().time_since_epoch();
+            const auto now_in_seconds = std::chrono::duration_cast<std::chrono::seconds>(now).count();
+
             Logger::info("Download progress: {}%", ctx.desiredState.downloadProgressPercentage);
-            send(ctx, IMqttProcessor::TOPIC_FEEDBACK, MqttMessage::Downloading);
+
+            if(now_in_seconds - _last_update > ctx.messageDelay) {
```

Comment:
shoud be operation >=:
now - last >= 5, 
e.g.   n s + 5 s =  (n+5) s  ==> send msg
(n+5) - n >= 5

Reply:
Fixed